### ISSUE TITLE
[Decaying] computeBasescore and tag_numerical_value_override: wrong score.

### ISSUE THREAD

**Issue by PakitoSec** [MISP/MISP#6352]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu
| OS version (client)       | Windows 10
| PHP version               | 7.2.
| MISP version / git hash   | 2.4.132
| Browser                   | Chrome

### Expected behavior
1. Set tag_numerical_value_override for tag: {"test:enriched": 90}
2. Create a new decay model and set base_score config with {"test": 1}
3. Simulate score on an event with the tag "test:enriched" : it should display Eff Ratio at 1 and Value at 90

### Actual behavior & Steps to reproduce the behavior
1. Set tag_numerical_value_override for tag: {"test:enriched": 90}
2. Create a new decay model and set base_score config with {"test": 1}
3. Simulate score: it says that there is no Eff. Ratio. but Value is 90.

### Resolution:
There is mapping error between tag and tax and the effective ratio can't be found.

In the file Base.php (DecayingModelsFormulas Models) line 109, you shouldn't use explode:
` $taxonomy = explode('=', $tag['Tag']['name'])[0];`
and use the existing function:
` $taxonomy = $this->__extractTagBasename($tag['Tag']['name'])['base'];`

Same here in app/webroot/js/decayingModelSimulation.js, line 551:
`var namespace = tag.Tag.name.split('=')[0];`
You should replace split with the equivalent __extractTagBasename php functions in javascript (**everywhere** in the javascript).

**Comment by mokaddem** [comment]
Hey,
Thanks a lot for the detailed issue and resolution.
Next time, do not hesitate to open a pull request!
Cheers

**Comment by PakitoSec** [comment]
@mokaddem : could you correct the javascript function _extractTagPart with the following code:
```
            _extractTagPart: function(tag) {
                var reTag = /^(?<namespace>[^:="]+):(?<predicate>[^:="]+)(="(?<value>[^:="]+)")?$/
                var result = tag.match(reTag);
                var tagBaseName = '';
                if (result !== null && result.groups.value !== undefined) {
                    tagBaseName = result.groups.namespace + ':' + result.groups.predicate;
                } else if (result !== null) {
                    tagBaseName = result.groups.namespace;
                } else {
                    tagBaseName = tag
                }
                return tagBaseName;
            }
```
Else, with current code, I have the following error: (for tag that don't have any namespace/predicate like: 'OTX')

```
VM858 decayingModelSimulation.js:647 Uncaught TypeError: Cannot read property 'groups' of null
    at BasescoreComputationTable._extractTagPart (VM858 decayingModelSimulation.js:647)
    at BasescoreComputationTable._get_computation_step (VM858 decayingModelSimulation.js:551)
    at Array.<anonymous> (VM858 decayingModelSimulation.js:584)
    at Array.d3_selectionPrototype.data (VM857 d3.js:896)
    at BasescoreComputationTable._draw (VM858 decayingModelSimulation.js:583)
    at BasescoreComputationTable.update (VM858 decayingModelSimulation.js:504)
    at Object.success (attribute_id:987:281)
    at fire (VM853 jquery.js:3496)
    at Object.fireWith [as resolveWith] (VM853 jquery.js:3626)
    at done (VM853 jquery.js:9786)
```

Next time I will pull request :)

**Comment by mokaddem** [comment]
@PakitoSec Hehe, looking forward to your PR!
(fixed in 40b3259b7a382977fc86ca49e68eed14baf076f8)
Thanks a lot!
